buildSrc/src/main/java/org/elasticsearch/gradle/testfixtures/TestFixturesPlugin.java

```diff
@@ -82,27 +104,54 @@ public void apply(Project project) {
             }
             tasks.withType(getTaskClass("com.carrotsearch.gradle.junit4.RandomizedTestingTask"), task ->
                 extension.fixtures.all(fixtureProject -> {
-                    task.dependsOn(fixtureProject.getTasks().getByName("composeUp"));
-                    task.finalizedBy(fixtureProject.getTasks().getByName("composeDown"));
-                    // Configure ports for the tests as system properties.
-                    // We only know these at execution time so we need to do it in doFirst
-                    task.doFirst(it ->
-                        fixtureProject.getExtensions().getByType(ComposeExtension.class).getServicesInfos()
-                            .forEach((service, infos) ->
-                                infos.getPorts()
-                                    .forEach((container, host) -> setSystemProperty(
-                                        it,
-                                        "test.fixtures." + fixtureProject.getName() + "." + service + "." + container,
-                                        host
-                                    ))
-                            ));
-                }));
+                    fixtureProject.getTasks().matching(it->it.getName().equals("buildFixture")).all(buildFixture ->
+                        task.dependsOn(buildFixture)
+                    );
+                    fixtureProject.getTasks().matching(it->it.getName().equals("composeDown")).all(composeDown ->
+                        task.finalizedBy(composeDown)
```

Comment:
I think we should have assurances this will occur, even if gradle gets a SIGINT. As a followup, we should spawn a child process that will catch sigint and shutdown this container (see similar work in the release manager).

Reply:
I think this is a bit different, since `./gradlew clean` cleans it up as `clean` depends on `composeDown`.  I was even contemplating not having this at all for this reason. 
The plugin and docker-compose are good at detecting and using existing containers and it's faster like that when you have to iterate many times on it. This is another way how this is different: new builds will not keep on adding containers.
Of course this would mean that the tests have to be written in a way that can handle that. 
I added the finalizedBy to make it resemble how it used to be but we'll need to consider how this will work with `--parallel` at some time. I propose we consider this then.

Reply:
So you think it is ok to leave the containers up? They certainly look like they are way lower overhead. So "best effort" seems not nearly so bad as it would be with containers.

Reply:
I don't think we should leave containers running. Whether or not they are rebuilt/reused is a different matter. But when the build finishes, regardless of whether it finished normally or was Ctrl-C'd, there should be nothing running (granted this is not exactly true because of the gradle daemon, but at least that is something that will destroy itself on a timer).

Reply:
OK, I'we put it on my TODO list to create a wrapper script to make sure containers are downed even if Gradle doesn't live long enough to run the finalizers.